ksql-api/src/main/java/io/confluent/ksql/api/server/ServerUtils.java

```diff
@@ -32,14 +37,29 @@ private ServerUtils() {
 
   public static void handleError(final HttpServerResponse response, final int statusCode,
       final int errorCode, final String errMsg) {
-    final JsonObject errResponse = new JsonObject().put("status", "error")
+    final JsonObject errResponse = createErrResponse(errorCode, errMsg);
+    response.setStatusCode(statusCode).end(errResponse.toBuffer());
+  }
+
+  public static JsonObject createErrResponse(final int errorCode, final String errMsg) {
+    return new JsonObject().put("status", "error")
         .put("errorCode", errorCode)
         .put("message", errMsg);
-    response.setStatusCode(statusCode).end(errResponse.toBuffer());
   }
 
   public static void unhandledExceptonHandler(final Throwable t) {
     log.error("Unhandled exception", t);
   }
 
+  public static JsonObject decodeJsonObject(final Buffer buffer,
+      final RoutingContext routingContext) {
+    try {
+      return new JsonObject(buffer);
+    } catch (DecodeException e) {
+      handleError(routingContext.response(), 400, ERROR_CODE_INVALID_JSON,
+          "Invalid JSON in request args");
```

Comment:
this isn't a particularly actionable message. can we improve on it by explaining why it's invalid?

Reply:
I will make sure we log the full information on the server, but I don't think it's a good idea to return the invalid JSON to the client (it could be huge for example)